src/core/network/injectNetworkStrategy.js

```diff
@@ -10,22 +10,29 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import injectSendXhrRequest from "./injectSendXhrRequest";
-import injectFetch from "./injectFetch";
-import injectSendBeacon from "./injectSendBeacon";
 import isFunction from "../../utils/isFunction";
 
-export default ({ window, logger }) => {
-  const fetch = isFunction(window.fetch)
-    ? injectFetch(window.fetch)
-    : injectSendXhrRequest(window.XMLHttpRequest);
-  const sendBeacon =
-    window.navigator && isFunction(window.navigator.sendBeacon)
-      ? injectSendBeacon(window.navigator, fetch, logger)
-      : fetch;
+export default ({
+  window,
+  logger,
+  injectSendFetchRequest,
+  injectSendXhrRequest,
+  injectSendBeaconRequest
+}) => {
+  const { fetch, navigator, XMLHttpRequest } = window;
+  const sendFetchRequest = isFunction(fetch)
+    ? injectSendFetchRequest({ fetch })
```

Comment:
I renamed `injectFetch` and `injectSendBeacon` to `injectSendFetchRequest` and `injectSendBeaconRequest` to distinguish their resulting functions (`sendFetchRequest` and `sendBeaconRequest` from their underlying native APIs (`fetch` and `sendBeacon`) and I also thought it would be nice if they were consistent with `injectSendXhrRequest`.